src/osd/ReplicatedPG.cc

```diff
@@ -8864,6 +8864,7 @@ void ReplicatedPG::on_change(ObjectStore::Transaction *t)
 
   // requeues waiting_for_active
   scrub_clear_state();
+  scrub_unreserve_replicas();
```

Comment:
Just took a look, scrub_clear_state is called by both primary and replica to clear the *local* scrub state, while scrub_unreserve_replicas only makes sense to primary. So we properly should keep the way it is? @liewegas